bftengine/src/communication/TlsTCPCommunication.cpp

```diff
@@ -956,28 +951,36 @@ class TlsTCPCommunication::TlsTcpImpl {
              uint16_t listenPort,
              uint32_t maxServerId,
              string listenIp,
-             string certRootFolder) :
+             string certRootFolder,
+             UPDATE_CONNECTIVITY_FN statusCallback = nullptr) :
       _selfId(selfNodeNum),
       _listenPort(listenPort),
       _listenIp(listenIp),
       _bufferLength(bufferLength),
       _maxServerId(maxServerId),
-      _certRootFolder(certRootFolder) {
+      _certRootFolder(certRootFolder),
+      _logger(Logger::getLogger("concord.tls")),
+      _statusCallback{statusCallback} {
     //_service = new io_service();
     for (auto it = nodes.begin(); it != nodes.end(); it++) {
       _nodes.insert({it->first, it->second});
     }
 
+    // all replicas are in listen mode
     if (_selfId < _maxServerId) {
       tcp::endpoint ep(address::from_string(_listenIp), _listenPort);
       _pAcceptor = boost::make_unique<tcp::acceptor>(_service, ep);
       start_accept();
-    } else LOG_INFO("skipping listen for node: " << _selfId);
+    } else // clients donn't listen
+      LOG_INFO(_logger, "skipping listen for node: " << _selfId);
 
+    // this node should connect only to nodes with lower ID
+    // and all nodes with higher ID will connect to this node
+    // we dont want that clients will connect to another clients
```

Comment:
nit: "we don't want clients to connection to other clients"

Reply:
nit: Neither "we do**nt** want **that** clients **will** connect to **an**other clients" nor "we don't want clients to connect**ion** to other clients" sounds completely right to me.